buildstockbatch/schemas/v0.3.yaml

```diff
@@ -19,11 +19,18 @@ postprocessing: include('postprocessing-spec', required=False)
 references: map(required=False)
 ---
 gcp-spec:
-  job_identifier: regex('^[a-zA-Z]\w{,9}$', required=True)
+  job_identifier: regex('^[a-zA-Z_-]\w{,20}$', required=True)
```

Comment:
Intentionally changing this to allow a job_identifier to start with hyphen and underscore? why?

Also, this is actually allowing the job_identifier to be 21 characters.

And what's the motivation of this change? I think we should avoid divergence with the AWS script unless necessary.

Reply:
Do we need to add the ability for it to start with an underscore? The current AWS regex allows underscores, just not as the first character.